### PR TITLE
Documentation: cross build of corefx for arm

### PR DESCRIPTION
**Pull request by leemgs**
Let's add a document how to do cross build of corefx for ARM[64]

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

### CODE REVIEW

**Comment by leemgs** [comment]
/cc  @jkotas, @benpye ,  @stephentoub
This patch is based on the previous PR https://github.com/dotnet/corefx/pull/6585

File: Documentation/building/cross-building.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+Cross Compilation for ARM on Linux
+==================================
+
+Through cross compilation, on Linux it is possible to build CoreFX for arm or arm64.
+It is almost similiar to the cross compilation procedure of CoreCLR.
```
Comment (stephentoub) [inline Documentation/building/cross-building.md]: Nit: almost similar => very similar

File: Documentation/building/cross-building.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+Generating the rootfs
+---------------------
+The `cross\build-rootfs.sh` script can be used to download the files needed for cross compilation. It will generate an Ubuntu 14.04 rootfs as this is what CoreFX targets.
+
+    Usage: build-rootfs.sh [BuildArch]
+    BuildArch can be: arm, arm64
+
+The `build-rootfs.sh` script must be run as root as it has to make some symlinks to the system, it will by default generate the rootfs in `cross\rootfs\<BuildArch>` however this can be changed by setting the `ROOTFS_DIR` environment variable.
```
Comment (stephentoub) [inline Documentation/building/cross-building.md]: Nit: "symlinks to the system, it" => "symlinks to the system. It"

**Comment by stephentoub** [comment]
This is very nice. Thanks for adding it. LGTM.

**Comment by benpye** [comment]
Is it correct to say that the managed binaries are OS and arch specific? OS
for Linux will be correct currently, but it is definitely different for
Windows and FreeBSD. I believe it's arch independent but would be worth
making sure, I don't think there is anything reliant on register length.

**Comment by stephentoub** [comment]
> Is it correct to say that the managed binaries are OS and arch specific?

They are architecture independent.

Many of the managed binaries are OS-independent, e.g. System.Linq.dll.  Some are OS-specific, e.g. System.IO.FileSystem.dll, with different builds for Windows vs Unix.  A few have different builds for different Unix flavors as well, e.g. System.Diagnostics.Process.dll has a different build for Linux vs OS X.

File: Documentation/building/cross-building.md (hunk 1 above)
Comment (leemgs) [inline Documentation/building/cross-building.md]: @stephentoub, Fixed. Thanks.

File: Documentation/building/cross-building.md (hunk 2 above)
Comment (leemgs) [inline Documentation/building/cross-building.md]: @stephentoub Fixed.

**Comment by leemgs** [comment]
@benpye, @stephentoub, :+1:   Updated wrong explanation.

**Pull request merged**